lib/ansible/cli/pull.py

```diff
@@ -37,7 +37,8 @@ class PullCLI(CLI):
         each set to run via cron and update playbook source via a source repository.
         This inverts the default *push* architecture of ansible into a *pull* architecture,
         which has near-limitless scaling potential.
-
+        To make concurrent ``ansible-pull`` invocations for the same checkout on the same host, you should
+        use either locks (``setlock/flock``) or separate directories.
```

Comment:
this is very specific to your situation, I would generalize this:

'None of the CLI tools are designed to run concurrently with themselves, you should use an external scheduler and/or locking to ensure there are no clashing operations'

Even running from a different directory might create a concurrency problem with 2 instances trying to modify the same resources.